src/main/java/org/spongepowered/common/SpongeImpl.java

```diff
@@ -197,18 +197,10 @@ public static boolean postEvent(Event event) {
         return Sponge.getEventManager().post(event);
     }
 
-    public static boolean postEvent(Event event, boolean allowClient) {
-        // TODO quick and dirty fix (cant cast in UnitTest)
-        if (Sponge.getEventManager() instanceof SpongeEventManager) {
-            return ((SpongeEventManager) Sponge.getEventManager()).post(event, allowClient);
-        }
-        return true;
-    }
-
     public static void postState(GameState state, GameStateEvent event) {
         check(game);
         game.setState(state);
-        ((SpongeEventManager) game.getEventManager()).post(event, true);
```

Comment:
The reason why this was added is because on clients, this will not work unless allowing a client thread event is true.